### PR TITLE
Added Keystone autentication

### PR DESCRIPTION
**Pull request by fguillot**
- Split API unit tests into multiple files
- Added pluggable authentication backend
- Added optional Keystone authentication (the default is the private key authentication backend which is stored in the config file)

### CODE REVIEW

File: almanach/common/exceptions/authentication_failure_exception.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (Marx314) [inline almanach/common/exceptions/authentication_failure_exception.py:15]: licences

**Comment by Marx314** [comment]
:+1: an integration test with keystone would be awesome but this PR is already big enough!

File: almanach/common/exceptions/authentication_failure_exception.py (hunk 1 above)
Comment (mbelang) [inline almanach/common/exceptions/authentication_failure_exception.py:15]: +1

Comment (fguillot) [inline almanach/common/exceptions/authentication_failure_exception.py:15]: The license header have been added guys.

Comment (mbelang) [inline almanach/common/exceptions/authentication_failure_exception.py:15]: This is wierd when I look at the the diff it was not there but it is there in the unified view...

**Comment by mbelang** [comment]
:+1:

**Comment by maxc0c0s** [comment]
👍

**Pull request merged**